File: app/src/main/java/com/example/laksh_pc/voteapp/navhome.java
package com.example.laksh_pc.voteapp;

import android.content.Intent;
import android.os.Bundle;
import android.util.Log;
import android.view.View;

import com.example.laksh_pc.voteapp.fingerPrint.Main2Activity;
import com.example.laksh_pc.voteapp.fingerPrint.MainActivity;
import com.google.android.material.navigation.NavigationView;
import com.google.firebase.auth.FirebaseAuth;
import com.google.firebase.database.DataSnapshot;
import com.google.firebase.database.DatabaseError;
import com.google.firebase.database.DatabaseReference;
import com.google.firebase.database.FirebaseDatabase;
import com.google.firebase.database.ValueEventListener;

import androidx.core.view.GravityCompat;
import androidx.drawerlayout.widget.DrawerLayout;
import androidx.appcompat.app.ActionBarDrawerToggle;
import androidx.appcompat.app.AppCompatActivity;
import androidx.appcompat.widget.Toolbar;
import android.view.Menu;
import android.view.MenuItem;
import android.widget.Button;
import android.widget.TextView;
import android.widget.Toast;

public class navhome extends AppCompatActivity
        implements NavigationView.OnNavigationItemSelectedListener {

    private TextView tvemail;
    private View includeview;
    private Button button8;
    String email="";
    private FirebaseAuth firebaseAuth;

    private DatabaseReference mdata1;
    private DatabaseReference mdata2;
    private DatabaseReference mdata3;
    private DatabaseReference mdata4;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_navhome);
        tvemail = (TextView) findViewById(R.id.dtext);
        tvemail.setText(getIntent().getExtras().getString("Email"));
        includeview=findViewById(R.id.contentView);
        button8 =includeview.findViewById(R.id.button8);
        firebaseAuth = FirebaseAuth.getInstance();
        email = firebaseAuth.getCurrentUser().getEmail().replace("@","").replace(".","");
        mdata1 = FirebaseDatabase.getInstance().getReference();
        mdata2 = FirebaseDatabase.getInstance().getReference();
        mdata3 = FirebaseDatabase.getInstance().getReference();
        mdata4 = FirebaseDatabase.getInstance().getReference();


        Toolbar toolbar = (Toolbar) findViewById(R.id.toolbar);
        setSupportActionBar(toolbar);
        String email=tvemail.getText().toString();
if (email.equals("admin@gmail.com")){
    button8.setVisibility(View.VISIBLE);
}else {
    button8.setVisibility(View.GONE);
}

        DrawerLayout drawer = (DrawerLayout) findViewById(R.id.drawer_layout);
        ActionBarDrawerToggle toggle = new ActionBarDrawerToggle(
                this, drawer, toolbar, R.string.navigation_drawer_open, R.string.navigation_drawer_close);
        drawer.setDrawerListener(toggle);
        toggle.syncState();

        NavigationView navigationView = (NavigationView) findViewById(R.id.nav_view);
        navigationView.setNavigationItemSelectedListener(this);
    }

    @Override
    public void onBackPressed() {
        DrawerLayout drawer = (DrawerLayout) findViewById(R.id.drawer_layout);
        if (drawer.isDrawerOpen(GravityCompat.START)) {
            drawer.closeDrawer(GravityCompat.START);
        } else {
            super.onBackPressed();
        }
    }

    @Override
    public boolean onCreateOptionsMenu(Menu menu) {
        // Inflate the menu; this adds items to the action bar if it is present.
        getMenuInflater().inflate(R.menu.navhome, menu);
        return true;
    }

    @Override
    public boolean onOptionsItemSelected(MenuItem item) {
        // Handle action bar item clicks here. The action bar will
        // automatically handle clicks on the Home/Up button, so long
        // as you specify a parent activity in AndroidManifest.xml.
        int id = item.getItemId();

        //noinspection SimplifiableIfStatement
        if (id == R.id.action_settings) {
            return true;
        }

        return super.onOptionsItemSelected(item);
    }

    @SuppressWarnings("StatementWithEmptyBody")
    @Override
    public boolean onNavigationItemSelected(MenuItem item) {
        // Handle navigation view item clicks here.
        int id = item.getItemId();

        if(id == R.id.nav_home){
            //noactivity
        }
        else if (id == R.id.nav_camera) {
            Intent g = new Intent(this,question.class);
            startActivity(g);

        } else if (id == R.id.nav_gallery) {
            Intent i = new Intent(this,profile.class);
            startActivity(i);

        } else if (id == R.id.nav_slideshow) {
            Intent k = new Intent(this,login.class);
            startActivity(k);
        }



        DrawerLayout drawer = (DrawerLayout) findViewById(R.id.drawer_layout);
        drawer.closeDrawer(GravityCompat.START);
        return true;
    }
    public void openlist(View view){

        final int[] count = {0};

        mdata1.addListenerForSingleValueEvent(new ValueEventListener() {
            @Override
            public void onDataChange(DataSnapshot dataSnapshot) {
                DataSnapshot size = dataSnapshot.child("Voted").child(email);
//                int output = Integer.parseInt(String.valueOf(size));
                if (size.getValue()==null){
                        Intent startnewactivity = new Intent(navhome.this, list.class);
                        startActivity(startnewactivity);
                    }else {
                    Log.e("TAG", "onDataChange: "+size.getValue().toString() );
                        Toast.makeText(getApplicationContext(),"you already vote",Toast.LENGTH_SHORT).show();
                    }

            }
//
            @Override
            public void onCancelled(DatabaseError databaseError) {

            }
        });
//        mdata2.addListenerForSingleValueEvent(new ValueEventListener() {
//            @Override
//            public void onDataChange(DataSnapshot dataSnapshot) {
//                long size1 = dataSnapshot.child("Hari").child(email).getChildrenCount();
//                int output = Integer.parseInt(String.valueOf(size1));
//                if (output>0){
//                    count[0] =output;
//                }
//
//            }
//
//            @Override
//            public void onCancelled(DatabaseError databaseError) {
//
//            }
//        });
//        mdata3.addListenerForSingleValueEvent(new ValueEventListener() {
//            @Override
//            public void onDataChange(DataSnapshot dataSnapshot) {
//                long size2 = dataSnapshot.child("Venkaiah Naidu").child(email).getChildrenCount();
//                int output = Integer.parseInt(String.valueOf(size2));
//                if (output>0){
//                    count[0] =output;
//                }
//
//            }
//
//            @Override
//            public void onCancelled(DatabaseError databaseError) {
//
//            }
//        });
//        mdata4.addListenerForSingleValueEvent(new ValueEventListener() {
//            @Override
//            public void onDataChange(DataSnapshot dataSnapshot) {
//                long size3 = dataSnapshot.child("Praveen").child(email).getChildrenCount();
//                int output = Integer.parseInt(String.valueOf(size3));
//                if (output>0){
//                    count[0] =output;
//                }
//            }
//
//            @Override
//            public void onCancelled(DatabaseError databaseError) {
//
//            }
//        });

    }
    public void live(View view){
        Intent g = new Intent(this, MainActivity.class);
        startActivity(g);
    }

}

File: app/src/main/java/com/example/laksh_pc/voteapp/Navhome.kt
package com.example.laksh_pc.voteapp

import android.content.Intent
import android.os.Bundle
import android.util.Log
import android.view.Menu
import android.view.MenuItem
import android.view.View
import android.widget.Button
import android.widget.TextView
import android.widget.Toast
import androidx.appcompat.app.ActionBarDrawerToggle
import androidx.appcompat.app.AppCompatActivity
import androidx.appcompat.widget.Toolbar
import androidx.core.view.GravityCompat
import androidx.drawerlayout.widget.DrawerLayout
import com.example.laksh_pc.voteapp.fingerPrint.MainActivity
import com.google.android.material.navigation.NavigationView
import com.google.firebase.auth.FirebaseAuth
import com.google.firebase.database.*

class Navhome : AppCompatActivity(), NavigationView.OnNavigationItemSelectedListener {
    private var tvemail: TextView? = null
    private var includeview: View? = null
    private var button8: Button? = null
    private var button5: Button? = null
    var email = ""
    private var firebaseAuth: FirebaseAuth? = null
    private var mdata1: DatabaseReference? = null
    private var mdata2: DatabaseReference? = null
    private var mdata3: DatabaseReference? = null
    private var mdata4: DatabaseReference? = null
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_navhome)
        tvemail = findViewById<View>(R.id.dtext) as TextView
        tvemail!!.text = intent.extras.getString("Email")
        includeview = findViewById(R.id.contentView)
        button8 = includeview!!.findViewById(R.id.button8)
        button5 = includeview!!.findViewById(R.id.button5)
        firebaseAuth = FirebaseAuth.getInstance()
        email = firebaseAuth!!.currentUser!!.email!!.replace("@", "").replace(".", "")
        mdata1 = FirebaseDatabase.getInstance().reference
        mdata2 = FirebaseDatabase.getInstance().reference
        mdata3 = FirebaseDatabase.getInstance().reference
        mdata4 = FirebaseDatabase.getInstance().reference
        val toolbar = findViewById<View>(R.id.toolbar) as Toolbar
        setSupportActionBar(toolbar)
        val email = tvemail!!.text.toString()
        if (email == "admin@gmail.com") {
            button8!!.setVisibility(View.VISIBLE)
            button5!!.setVisibility(View.VISIBLE)
        } else {
            button8!!.setVisibility(View.GONE)
            button5!!.setVisibility(View.GONE)
        }
        val drawer = findViewById<View>(R.id.drawer_layout) as DrawerLayout
        val toggle = ActionBarDrawerToggle(
                this, drawer, toolbar, R.string.navigation_drawer_open, R.string.navigation_drawer_close)
        drawer.setDrawerListener(toggle)
        toggle.syncState()
        val navigationView = findViewById<View>(R.id.nav_view) as NavigationView
        navigationView.setNavigationItemSelectedListener(this)
    }

    override fun onBackPressed() {
        val drawer = findViewById<View>(R.id.drawer_layout) as DrawerLayout
        if (drawer.isDrawerOpen(GravityCompat.START)) {
            drawer.closeDrawer(GravityCompat.START)
        } else {
            super.onBackPressed()
        }
    }

    override fun onCreateOptionsMenu(menu: Menu): Boolean { // Inflate the menu; this adds items to the action bar if it is present.
        menuInflater.inflate(R.menu.navhome, menu)
        return true
    }

    override fun onOptionsItemSelected(item: MenuItem): Boolean { // Handle action bar item clicks here. The action bar will
// automatically handle clicks on the Home/Up button, so long
// as you specify a parent activity in AndroidManifest.xml.
        val id = item.itemId
        return if (id == R.id.action_settings) {
            true
        } else super.onOptionsItemSelected(item)
    }

    override fun onNavigationItemSelected(item: MenuItem): Boolean { // Handle navigation view item clicks here.
        val id = item.itemId
        if (id == R.id.nav_home) { //noactivity
        } else if (id == R.id.nav_camera) {
            val g = Intent(this, Question::class.java)
            startActivity(g)
        } else if (id == R.id.nav_gallery) {
            val i = Intent(this, Profile::class.java)
            startActivity(i)
        } else if (id == R.id.nav_slideshow) {
            val k = Intent(this, Login::class.java)
            startActivity(k)
        }
        val drawer = findViewById<View>(R.id.drawer_layout) as DrawerLayout
        drawer.closeDrawer(GravityCompat.START)
        return true
    }

    fun openlist(view: View?) {
        val count = intArrayOf(0)
        mdata1!!.addListenerForSingleValueEvent(object : ValueEventListener {
            override fun onDataChange(dataSnapshot: DataSnapshot) {
                val size = dataSnapshot.child("Voted").child(email)
                //                int output = Integer.parseInt(String.valueOf(size));
                Log.e("TAG", "onDataChange: " + size.value)
                if (size.value == null) {
                    val startnewactivity = Intent(this@Navhome, List::class.java)
                    startActivity(startnewactivity)
//                            finish()
                } else {
                    Log.e("TAG", "onDataChange: " + size.value.toString())
                    Toast.makeText(applicationContext, "you already vote", Toast.LENGTH_SHORT).show()
                }
            }

            //
            override fun onCancelled(databaseError: DatabaseError) {}
        })
        //        mdata2.addListenerForSingleValueEvent(new ValueEventListener() {
//            @Override
//            public void onDataChange(DataSnapshot dataSnapshot) {
//                long size1 = dataSnapshot.child("Hari").child(email).getChildrenCount();
//                int output = Integer.parseInt(String.valueOf(size1));
//                if (output>0){
//                    count[0] =output;
//                }
//
//            }
//
//            @Override
//            public void onCancelled(DatabaseError databaseError) {
//
//            }
//        });
//        mdata3.addListenerForSingleValueEvent(new ValueEventListener() {
//            @Override
//            public void onDataChange(DataSnapshot dataSnapshot) {
//                long size2 = dataSnapshot.child("Venkaiah Naidu").child(email).getChildrenCount();
//                int output = Integer.parseInt(String.valueOf(size2));
//                if (output>0){
//                    count[0] =output;
//                }
//
//            }
//
//            @Override
//            public void onCancelled(DatabaseError databaseError) {
//
//            }
//        });
//        mdata4.addListenerForSingleValueEvent(new ValueEventListener() {
//            @Override
//            public void onDataChange(DataSnapshot dataSnapshot) {
//                long size3 = dataSnapshot.child("Praveen").child(email).getChildrenCount();
//                int output = Integer.parseInt(String.valueOf(size3));
//                if (output>0){
//                    count[0] =output;
//                }
//            }
//
//            @Override
//            public void onCancelled(DatabaseError databaseError) {
//
//            }
//        });
    }

    public fun clickNominate(v: View?){
        val startnewactivity = Intent(this@Navhome, UserListFragment::class.java)
        startActivity(startnewactivity)
//        finish()
    }

    fun live(view: View?) {
        val g = Intent(this, MainActivity::class.java)
        startActivity(g)
    }
}